lib/miq_automation_engine/service_models/miq_ae_service_physical_server_provision_task.rb

```diff
@@ -0,0 +1,4 @@
+module MiqAeMethodService
+  class MiqAeServicePhysicalServerProvisionTask < MiqAeServiceMiqProvisionTask
+  end
+end
```

Comment:
Is this file necessary? @gmcculloug

Reply:
No, automate will dynamically create service models and expose columns, virtual_columns and associations from the active_record model.  You only need to create these models now if you modify or expose additional functionality.

Reply:
Which I do - I extend `MiqAeServiceMiqProvisionTask` which have a bunch of custom methods like state_machine_status and set_option() that we need.

Reply:
The point being this model is empty so those methods are either being exposed automatically or pulled in from sub-classes.  If there is nothing defined here the empty class is not needed.

Reply:
I think it didn't work until I added the definition here because [this method](https://github.com/ManageIQ/manageiq-automation_engine/blob/master/lib/miq_automation_engine/service_models/miq_ae_service_miq_provision_task.rb#L3) was not defined  but let me try again tomorrow, probably I messed something up. Anyhow, thanks for pointing up, I had no idea service models are able to automatically inherit between themselves based on actual model relations.

Reply:
If you are interested in learning more check out https://github.com/ManageIQ/manageiq/pull/14465 and https://github.com/ManageIQ/manageiq-automation_engine/pull/308

Reply:
Surprisingly, you were right again 👍 Works like a charm without the file. Thanks for pointers to PRs